NethermindNode.Tests/Tests/Pruning/JsonRpcPruning.cs

```diff
@@ -69,31 +76,43 @@ public void ShouldPruneDbUsingAdminRpc()
                 "Full Pruning Finished"
             };
 
-            int expectedLogIndex = 0;
+            HashSet<string> missingLogs = new HashSet<string>(expectedLogs);
 
             try
             {
-                foreach (var line in DockerCommands.GetDockerLogs("sedge-execution-client", "Full Pruning", true, cts.Token))
+                foreach (var line in DockerCommands.GetDockerLogs("sedge-execution-client", "Full Pruning", true, cts.Token, "--since 2m")) //since to ensure that we will get only recent logs but including all from beggining of test
                 {
-                    Console.WriteLine(line);
-
-                    if (line.Contains(expectedLogs[expectedLogIndex]))
-                    {
-                        expectedLogIndex++;
-                    }
+                    Console.WriteLine(line); // For visibility during testing
 
-                    if (expectedLogIndex >= expectedLogs.Length)
+                    foreach (var expectedLog in missingLogs)
                     {
-                        break;
+                        if (line.Contains(expectedLog))
+                        {
+                            Logger.Info($"Log found: \"{line}\" - Expected log: {expectedLog}");
+                            missingLogs.Remove(expectedLog);
+
+                            if (expectedLog == expectedLogs.Last())
+                            {
+                                // End becuase Pruning itself may work but for some reason some logs may not be found - so better this way than waiting for all
+                                cts.Cancel();
+                                break;
+                            }
+                        }
                     }
                 }
             }
             catch (OperationCanceledException)
             {
-                Console.WriteLine("Operation was canceled."); 
+                Logger.Info("Operation was canceled.");
+            }
+
+            if (missingLogs.Count > 0)
+            {
+                // In case some of the logs were not displayed log Warning
+                Logger.Warn($"Missing logs: {string.Join(", ", missingLogs)}");
             }
 
-            Assert.That(expectedLogIndex, Is.EqualTo(expectedLogs.Length), "Not all expected log substrings were found in order.");
+            Assert.That(missingLogs.Count, Is.EqualTo(0), "Not all expected log substrings were found.");
```

Comment:
Not sure if the logs are included in the test output, but it may be a good idea to mention missing logs in this assert.

Reply:
Yeah those are - would improve this